sklearn/inspection/tests/test_permutation_importance_polars.py

```diff
@@ -0,0 +1,50 @@
+"""Tests for permutation_importance with Polars DataFrames."""
+
+import numpy as np
+import pytest
+
+from sklearn import set_config
+from sklearn.compose import ColumnTransformer
+from sklearn.inspection import permutation_importance
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import Pipeline
+from sklearn.preprocessing import StandardScaler
+
+pl = pytest.importorskip("polars")
+
+
+def test_permutation_importance_polars_dataframe():
+    """Test that permutation_importance works with Polars DataFrames."""
+    set_config(transform_output="polars")
```

Comment:
Instead of changing the global config state in a test with `set_config`, please wrap the lines of the test that needs it under a context manager:

```python
with sklearn.config_context(transform_output="polars"):
    ...
```

This way, the config state is restored to its default value after the execution of those lines.